underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -386,6 +386,7 @@ public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
             setOwner(dirToMake.toString(), options.getOwner(), options.getGroup());
           } catch (IOException e) {
             LOG.warn("Failed to update the ufs dir ownership, default values will be used. " + e);
+            throw e;
```

Comment:
@maobaolong I think it is intended to not throw the exception here.

As the comment described, it is possible that Alluxio user does not have permission to chown on the UFS. E.g. some user is testing Alluxio on a local machine and Alluxio is not running under root user. In this case, we don't want to fail the chown op, so we only print out warning instead.

Reply:
@ifcharming I see. If the testing user is not exist on a testing local ufs, Although we use root user to execute the test, the exception will occurr too.

Reply:
yeah, mocking the local user SGTM, thanks

Reply:
@ifcharming if `setOwner` fails, then the file in UFS will not be in sync with Alluxio as the owner will be different. Do you think this is acceptable from the correctness? Even if we want to suppress the exception, I feel the UFS should throw the exception and then return it from the worker, so the client can log the warning on the client-side.

Reply:
@ifcharming In consideration of we want to add the condition of owner consistent to check consistent of inode and ufs. So now we should not tolerate this exception, instead, we must throw it. 

And as deal, we must mocking the local user to let the failed unit test passed, and we must fix the test first, in case pending PR like this.

Reply:
@maobaolong can you wrap this in the condition of `PropertyKey.UNDERFS_ALLOW_SET_OWNER_FAILURE)` as done in `setOwner` method? So we will throw the exception unless the user allow the set-owner failure.